C error when trying to unwrap a module constant, optional string to an int

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Pop!_OS 21.10
Processor: 6 cpus, 64bit, little endian, AMD Ryzen 5 4500U with Radeon Graphics
CC version: cc (Ubuntu 11.2.0-7ubuntu2) 11.2.0

getwd: /home/astro/Research/Thesis/rng_testbench
vmodules: /home/astro/.vmodules
vroot: /home/astro/Projects/V/v-main
vexe: /home/astro/Projects/V/v-main/v
vexe mtime: 2022-04-03 08:47:35
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 10ab758.0497b88

Git version: git version 2.32.0
Git vroot status: weekly.2022.13-40-g0497b885
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 3654d6a8

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg old/scratch.v`
```v
module main

import os

const iterations = (os.getenv_opt('ITERATIONS') or { '5' }).int()

fn main() {
	println('Number of iterations: $iterations')
}

```

**What did you expect to see?**

Number of iterations: 5


**What did you see instead?**
```
==================
/tmp/v_1000/scratch.7801443355781935512.tmp.c:16132: error: cannot convert 'struct Option_string' to 'struct string'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

**Note**

If we omit the `.int()`, it works.

```
const iterations = (os.getenv_opt('ITERATIONS') or { '5' })
```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -121,6 +121,7 @@ mut:
 	inside_comptime_for_field bool
 	inside_cast_in_heap       int // inside cast to interface type in heap (resolve recursive calls)
 	inside_const              bool
+	inside_const_optional     bool
 	inside_lambda             bool
 	loop_depth                int
 	ternary_names             map[string]string
@@ -4174,11 +4175,13 @@ fn (mut g Gen) const_decl(node ast.ConstDecl) {
 			}
 			ast.CallExpr {
 				if field.expr.return_type.has_flag(.optional) {
+					g.inside_const_optional = true
 					unwrap_option := field.expr.or_block.kind != .absent
 					g.const_decl_init_later(field.mod, name, field.expr, field.typ, unwrap_option)
 				} else {
 					g.const_decl_init_later(field.mod, name, field.expr, field.typ, false)
 				}
+				g.inside_const_optional = false
 			}
 			else {
 				// Note: -usecache uses prebuilt modules, each compiled with:

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -698,7 +698,7 @@ fn (mut g Gen) call_expr(node ast.CallExpr) {
 		} else if g.table.sym(node.return_type).kind == .multi_return {
 			g.write('\n $cur_line $tmp_opt /*U*/')
 		} else {
-			if !g.inside_const {
+			if !g.inside_const || !g.inside_const_optional {
 				g.write('\n $cur_line (*($unwrapped_styp*)${tmp_opt}.data)')
 			} else {
 				g.write('\n $cur_line $tmp_opt')

--- a/vlib/v/tests/const_use_nested_optionals_test.v
+++ b/vlib/v/tests/const_use_nested_optionals_test.v
@@ -1,0 +1,10 @@
+module main
+
+import os
+
+const iterations = (os.getenv_opt('ITERATIONS') or { '5' }).int()
+
+fn test_const_use_nested_optionals() {
+	println('Number of iterations: $iterations')
+	assert iterations == 5
+}

